airflow/bin/cli.py

```diff
@@ -607,6 +636,10 @@ class CLIFactory(object):
         'execution_date': Arg(
             ("execution_date",), help="The execution date of the DAG",
             type=parsedate),
+        'execution_date_enhanced': Arg(
```

Comment:
You can remove the `type` argument in `execution_date` and take charge. That would ensure consistent behavior of execution_date across the CLI.